src/client/ua_client_connect.c

```diff
@@ -1709,6 +1709,72 @@ UA_Client_connectSecureChannel(UA_Client *client, const char *endpointUrl) {
     return client->connectStatus;
 }
 
+static void
+activateSessionSync(UA_Client *client) {
+    UA_LOCK_ASSERT(&client->clientMutex, 1);
+    
+    UA_DateTime now = UA_DateTime_nowMonotonic();
+    UA_DateTime maxDate = now + ((UA_DateTime)client->config.timeout * UA_DATETIME_MSEC);
+
+    // reset state to created
+    client->sessionState = UA_SESSIONSTATE_CREATED;
+    activateSessionAsync(client);
+    notifyClientState(client);
+
+    /* EventLoop is started. Otherwise activateSessionSync would have failed. */
+    UA_EventLoop *el = client->config.eventLoop;
+    UA_assert(el);
+    while(true) {
+
+        if(client->connectStatus != UA_STATUSCODE_GOOD &&
```

Comment:
Just abort if the connectStatus is not good.
We cannot recover from that. Irrespective of the channel state.